ssl/s3_lib.c

```diff
@@ -4723,12 +4736,26 @@ EVP_PKEY *ssl_generate_pkey_group(SSL *s, uint16_t id)
                  ERR_R_EVP_LIB);
         goto err;
     }
-    if (gtype != TLS_CURVE_CUSTOM
+# ifndef OPENSSL_NO_DH
+    if (gtype == TLS_GROUP_FFDHE
+            && (!(pkey = EVP_PKEY_new())
+            || !EVP_PKEY_assign(pkey, EVP_PKEY_DH, DH_new_by_nid(ginf->nid))
+            || EVP_PKEY_CTX_set_dh_nid(pctx, ginf->nid) <= 0)) {
+        SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_SSL_GENERATE_PKEY_GROUP,
+                ERR_R_EVP_LIB);
+        EVP_PKEY_free(pkey);
+        pkey = NULL;
+        goto err;
+    }
+# endif
+# ifndef OPENSSL_NO_EC
+    if (ecgtype && ecgtype != TLS_GROUP_CURVE_CUSTOM
```

Comment:
why the test against zero?  Seems redundant.

Reply:
Here `ecgtype` will be 0 incase of FFDHE key exchange, so that `ecgtype != 0` check is needed here.